### PR TITLE
Add suggestions to doc to look in make/local

### PR DESCRIPTION
**Pull request by seantalts**
And remove unused CC= declarations etc in the makefile that were
confusing people.

#### Submisison Checklist

- [x] Run tests: `./runCmdStanTests.py src/test`
- [x] Declare copyright holder and open-source license: see below

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Columbia University

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -11,16 +11,13 @@ help:
 
 ##
 # Users should only need to set these three variables for use.
+# Please set them in a test file called make/local with no extension.
 # - CC: The compiler to use. Expecting g++ or clang++.
```
Comment (syclik) [inline makefile]: Change `CC` to `CXX`.

File: src/docs/cmdstan-guide/getting-started.tex [hunk 2]
```diff
@@ -910,7 +910,8 @@ \subsection{Configuring Command-Line Options}
 %% \emph{Warning}: \ The unit tests can take 30+ minutes and consume 3+
 %% GB of memory with the default compiler, \code{g++}.  The distribution
 %% test and model tests can take even longer.  It is faster to run the
-%% Clang compiler (option \code{CC=clang++}), and to run in multiple
+%% Clang compiler (option \code{CC=clang++} in make/local), and to run
```
Comment (syclik) [inline src/docs/cmdstan-guide/getting-started.tex]: change `CC` to `CXX`

**Comment by syclik** [comment]
I don't think this actually fixes the issues with the builds. We should take care of that before releasing the next version.

**Comment by seantalts** [comment]
What build issues are you referring to specifically?

And I think it’s supposed to be CC, actually, looking at how it’s used?

**Comment by sakrejda** [comment]
Libc++flag contaminates builds that switch to GCC via CC variable, for
example

**Comment by seantalts** [comment]
Oh, like people wanted to be able to compile more easily with stdlib instead of libc++?

**Comment by sakrejda** [comment]
Not quite, the libc++ flag should be compiler-specific so if we're not
using clang++ it should not get pulled in at all.

**Comment by seantalts** [comment]
There are major LTS versions of Ubuntu (used in Travis) with broken stdlib but working libc++, I think :( I think letting users set it is probably also important for that reason. Would probably make more sense to have the default library be the one that comes with the compiler you're using though. Let me open an issue for that.

**Comment by sakrejda** [comment]
Yeah, I know, also versions of Ubuntu that can't successfully install clang++.... my point is we should make it straightforward to change these in ways that allow all working combos.  I tried to do this but I ended up making changes across math/stan/cmdstan makefiles and never backed out what the actual commits would need to be.

**Comment by seantalts** [comment]
Gotcha. Thanks for giving it a shot! I think I can do a simple patch and await more ambitious refactorings later.

**Comment by seantalts** [comment]
https://github.com/stan-dev/cmdstan/issues/590

**Pull request closed without merging**